### PR TITLE
Update variable mapping

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable) (N/A)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

### CODE REVIEW

File: tests/test_definitions.py [hunk 1]
```diff
@@ -92,12 +104,23 @@ def test_convert_magicc7_to_openscm_variables(magicc7, openscm):
     [
         ("HFC245FA_EMIS", "Emissions|HFC245fa"),
         ("CO2I_EMIS", "Emissions|CO2|MAGICC Fossil and Industrial"),
-        ("CH4_EMIS", "Emissions|CH4"),
         ("CH3CCL3_EMIS", "Emissions|CH3CCl3"),
         ("CCL4_INVERSE_EMIS", "Inverse Emissions|CCl4"),
+        ("CO2_EMIS", "Emissions|CO2"),
```
Comment (lewisjared) [inline tests/test_definitions.py]: Should this be CO2T_EMIS? In the source there is no reference to a CO2_EMIS variable

File: pymagicc/definitions/__init__.py [hunk 2]
```diff
@@ -283,6 +285,8 @@ def get_openscm_replacement(in_var):
             )
         elif variable.endswith("B") and not edge_case_B:
             variable = DATA_HIERARCHY_SEPARATOR.join([variable[:-1], "MAGICC AFOLU"])
+        elif variable.endswith("T") and not edge_case_B:
```
Comment (lewisjared) [inline pymagicc/definitions/__init__.py]:
```suggestion
        elif variable.endswith("T"):
```

**Comment by znicholls** [comment]
> Does it handle `DAT_SURFACE_TEMP_SUBANNUAL` or `*_GWPEMIS`?

`DAT_SURFACE_TEMP_SUBANNUAL` nope but will add. I'd rather not support `*_GWPEMIS`, I think this sort of conversion should be done in post-processing rather than relying on the values hard-coded in MAGICC to be right...

File: tests/test_definitions.py (hunk 1 above)
Comment (znicholls) [reply tests/test_definitions.py]: I think it doesn't actually matter as MAGICC has to have a fossil/afolu split (i.e. you could never prescribe total CO2 emissions). In any case, I have followed the convention for methane. In the output, `CH4T_EMIS` is total methane emissions but in the input of a SCEN7 file, the column header has to be `CH4_EMIS`. Hence the mapping has to be different depending on whether you're going forwards or backwards.

**Pull request merged**